src/thread_task.rs

```diff
@@ -774,74 +761,74 @@ unsafe fn delayed_fg_task(c: *const Rav1dContext, ttd: *mut TaskThreadData) {
             }
         }
         row = ::core::intrinsics::atomic_xadd_seqcst(
-            &mut *((*ttd).delayed_fg.progress).as_mut_ptr().offset(0) as *mut atomic_int,
+            &mut *(ttd.delayed_fg.progress).as_mut_ptr().offset(0) as *mut atomic_int,
             1 as c_int,
         );
         #[allow(unused_assignments)]
         // TODO(kkysen) non-trivial due to the atomics, so leaving for later
         {
             done = ::core::intrinsics::atomic_xadd_seqcst(
-                &mut *((*ttd).delayed_fg.progress).as_mut_ptr().offset(1) as *mut atomic_int,
+                &mut *(ttd.delayed_fg.progress).as_mut_ptr().offset(1) as *mut atomic_int,
                 1 as c_int,
             ) + 1;
         }
         if row < progmax {
             continue;
         }
-        pthread_mutex_lock(&mut (*ttd).lock);
-        (*ttd).delayed_fg.exec = 0 as c_int;
+        pthread_mutex_lock(&mut ttd.lock);
+        ttd.delayed_fg.exec = 0 as c_int;
         break;
     }
     done = ::core::intrinsics::atomic_xadd_seqcst(
-        &mut *((*ttd).delayed_fg.progress).as_mut_ptr().offset(1) as *mut atomic_int,
+        &mut *(ttd.delayed_fg.progress).as_mut_ptr().offset(1) as *mut atomic_int,
         1 as c_int,
     ) + 1;
     progmax = ::core::intrinsics::atomic_load_seqcst(
-        &mut *((*ttd).delayed_fg.progress).as_mut_ptr().offset(0) as *mut atomic_int,
+        &mut *(ttd.delayed_fg.progress).as_mut_ptr().offset(0) as *mut atomic_int,
     );
     if !(done < progmax) {
-        pthread_cond_signal(&mut (*ttd).delayed_fg.cond);
+        pthread_cond_signal(&mut ttd.delayed_fg.cond);
     }
 }
 
 pub unsafe extern "C" fn rav1d_worker_task(data: *mut c_void) -> *mut c_void {
-    let tc: *mut Rav1dTaskContext = data as *mut Rav1dTaskContext;
-    let c: *const Rav1dContext = (*tc).c;
-    let ttd: *mut TaskThreadData = (*tc).task_thread.ttd;
+    let tc: &mut Rav1dTaskContext = (data as *mut Rav1dTaskContext).as_mut().unwrap();
```

Comment:
Can this just be a `&mut *` like we do elsewhere?  Once we replace the thread spawning with the Rust version we can pass the data directly so I don't think the unwrap is helpful.